File: Python/_1846_Maximum_Element_After_Decreasing_and_Rearranging/Solution.py
# coding=utf-8
# Date: 2021/7/15 10:06
from typing import List


# 执行用时：92 ms, 在所有 Python3 提交中击败了88.73%的用户
# 内存消耗：21.6 MB, 在所有 Python3 提交中击败了21.53%的用户
class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        n = len(arr)
        arr.sort()
        arr[0] = 1
        for i in range(1, n):
            if arr[i] > arr[i - 1] + 1:
                arr[i] = arr[i - 1] + 1
        return arr[n - 1]


if __name__ == "__main__":
    print(Solution().maximumElementAfterDecrementingAndRearranging([2, 2, 1, 2, 1]))  # 2

File: Java/src/_1846_Maximum_Element_After_Decreasing_and_Rearranging/Solution.java
package _1846_Maximum_Element_After_Decreasing_and_Rearranging;

import java.util.Arrays;

/**
 * Date: 2021/7/15 10:00
 * Content:
 */
public class Solution {

    public int maximumElementAfterDecrementingAndRearranging(int[] arr) {
        int n = arr.length;
        Arrays.sort(arr);
        arr[0] = 1;
        for (int i = 1; i < n; i++)
            if (arr[i] > arr[i - 1] + 1)
                arr[i] = arr[i - 1] + 1;
        return arr[n - 1];
    }

    public static void main(String[] args) {

        int[] arr = {2, 2, 1, 2, 1};
        System.out.println((new Solution()).maximumElementAfterDecrementingAndRearranging(arr));  // 2
    }
}
